app/actions/QuoteActions.js

```diff
@@ -39,6 +39,9 @@ export const fetchAll =
     if (!endpoint) {
       return Promise.resolve(null);
     }
+    if (!loadNextPage) {
+      dispatch({ type: Quote.CLEAR });
+    }
```

Comment:
When we navigate from another part of the page, this will clear the quote state regardless of the fact that the requested items have already been loaded. I.e. this will fix the edgecase you want, but at the same time, it will cause the page to load every time we navigate from somewhere to `/quotes`.

You can probably see this if you go frontpage -> /quotes -> <somewhere else, f.ex /events> -> /quotes.
In the current version, the page won't load in the last step.
With this change, it will load everything again.

Reply:
Your mentioned issue is the behaviour I expect when re-entering a list view. Every time I click on "Overhørt" again after being somewhere else and get routed to the middle of the previously loaded list of quotes, I am surprised and may sometimes reload the page. To me, going away from the list view of quotes and re-entering it, is like the user telling us "please update the list of quotes". Thus, I do not see this as an issue. 

This is, of course, just my opinion. If you want to keep the current behaviour where all the quotes are kept in redux while you navigate other parts of the website, I think there are two alternatives:
1. As you mentioned, to add a flag to each randomly retrieved quote and only clear the quotes with this flag. 
2. To make the random quote not be added to the redux state. This does, however, conflict with the structure where every request goes through redux.

I would like your answer before any more changes are made. Should we keep it as I suggested, add the flag and filter (1), or remove random quotes from redux (2). I think (2) is best, but it conflicts with the rest of the project structure.  

The overarching problem is that most API requests on the website are going through redux and are added to the global redux state. This makes the code overcomplicated and creates problems like this. However, fixing the redux problem will take a lot more time.

Reply:
We discussed this at our meeting, and think that it could be a viable solution to make a separate state for random Quotes in Redux. This would be in line with our architecture, even though it is a bit excessive. (Since it is only used for one component) What do you think about this approach, @ulrik2204 ?

Reply:
You could also use the same top-level `quote` state, and store the random quote under another key. So you just modify the reducer to store it here, and modify the selector to use this as well. In this case you don't need to create a whole new entityReducer and everything.

Reply:
Yeah, using another key seems like the best option. I'll see what I can do.